tests/ErrorHandlerTest.php

```diff
@@ -358,6 +278,50 @@ public function testHandleExceptionWithThrowingPreviousExceptionHandler(): void
         } finally {
             restore_error_handler();
             restore_exception_handler();
+
+            $this->assertSame(2, $listenerCalled);
+            $this->assertSame($exception1, $captured1);
+            $this->assertSame($exception2, $captured2);
         }
     }
+
+    /**
+     * @dataProvider callableProvider
+     */
+    public function testAddListener(callable $listener): void
+    {
+        $exception = new \Exception();
+
+        try {
+            ErrorHandler::addExceptionListener($listener);
+
+            ErrorHandler::registerOnce()->handleException($exception);
+        } catch (\Throwable $rethrownException) {
+            $this->assertSame($exception, $rethrownException);
+        } finally {
+            restore_error_handler();
+            restore_exception_handler();
+        }
+    }
+
+    public function callableProvider(): array
```

Comment:
I'm not sure it's worth to have and maintain this test, I would just use `call_user_func` all over the codebase and we're sure that if something that's an incorrect `callable` is passed to the functions it's user's fault and not our

Reply:
I can understand the maintenance cost objection. I've changed the test to check for exceptions from listeners.